### ISSUE TITLE
Widget for Unsplash integration

### ISSUE THREAD

**Issue by love2dishtech** [netlify/netlify-cms#2579]
**Is your feature request related to a problem? Please describe.**
I have moved to SSG(Gatsby) from Ghost.org and Medium.com and one of there extremely nice feature with authoring was the idea of linking images from Unsplash. I believe this is one of the key features that a blog developer look at, especially because Unsplash provides free to use(with some attribution) content for the blogs/sites. I would usually use them for Cover/Featured image and sometimes in the main content as well. It is extremely irritating and time consuming process to find an image which can be reused in our site without much copyright issues. Finding/Downloading and then Uploading to our repo is a lengthy and irritating process.

**Describe the solution you'd like**
The way it worked in Ghost.org and Medium.com is pretty awesome. As a content writer we have an option to simply click a button that opens up a widget for exploring an image from thousands of Unsplash images.

A widget for Unsplash image can be really helpful that allows the writer to explore the images out of the box from the CMS.

I would expect it to add the public URL and attribution text to the markdown frontmatter, where it can be picked by the site.

**Describe alternatives you've considered**
The alternatives currently I am exploring is to manually go to Unsplash website and explore images. I would then look for the public URL and add it to a String URL in the content. Similarly, I would also look for attribution link and add it as text for the frontmatter.

**Additional context**
[Gatsby Integration Blog](https://medium.com/unsplash/a-few-of-unsplashs-api-partners-95e4bdc79870)

**Comment by erquhart** [comment]
An unsplash widget would be awesome, would love to see that.

**Comment by love2dishtech** [comment]
This is so cool, that this feature is accepted. It would be nice to see something like this. One addition, I would add in the existing alternative is that the Unsplash APIs terms require the widget to invoke an additional download call to the Unsplash APIs in case of there APIs. This helps them to identify if the image has been downloaded or not.

**Comment by austincondiff** [comment]
Why stop at Unsplash? What if we allowed developers to include whatever media source either by way of configuration with built-in support or by extension? 

To build upon #2557, it might look something like this...
![Editor View - No Image - Choose Image - Sources](https://user-images.githubusercontent.com/806104/65289461-5398e700-db10-11e9-8bd9-37dc95c7538b.jpg)

**Comment by love2dishtech** [comment]
That would be really great, especially considering that currently it requires explicit coding and processing to have multiple optional sources. As an example, in my current project, I am optionally using a feature image source from Unsplash and another from a local source. And I can see that some of my sources could be coming from Flickr and so.

However, that brings an important design consideration. How should we handle different configurations for required widget behaviour?

To give an example, for Unsplash to work fine with the backend system, I am capturing the raw image URLs which does not include any of the default generated ID's and transformations and similarly for cloudinary sources I am using only the public id, or again the raw URL without any transformations. This all to allow all the custom transformations in the gatsby source. However, for widget to be generic enough, it should have the capacity to either extract transformed images or raw images.

**Comment by erquhart** [comment]
Good point, integrating Unsplash with the media library makes more sense than having a dedicated Unsplash widget. Definitely worth discussing.

**Comment by tinfoil-knight** [comment]
@austincondiff Can I take this issue?

**Comment by erezrokah** [comment]
@tinfoil-knight are you interested in doing this for v2 or for the new UX/UI effort?

**Comment by tinfoil-knight** [comment]
@erezrokah I was interested in the new UI/UX effort.

**Comment by erezrokah** [comment]
I think it is better to do it under the new UI/UX effort.

**Comment by tinfoil-knight** [comment]
Current Work Done on `v3-ui-redesign` branch:

`MediaDialog` component has been created

<img width="857" alt="Screenshot 2020-07-27 at 2 55 51 PM" src="https://user-images.githubusercontent.com/51631122/88526233-4e060a80-d019-11ea-9eb2-a62d5c1bde52.png">

Ref:
6d2f19b4d2fdf7dc5873549d82e09a51869477fa
6d2f19b4d2fdf7dc5873549d82e09a51869477fa

**Comment by tinfoil-knight** [comment]
@erezrokah Would the images from external source(s) (like Unsplash) be copied over to the user's repo or would only the links be maintained?

**Comment by FR073N** [comment]
This feature would be really awesome. Any news since last state @tinfoil-knight ?

**Comment by tinfoil-knight** [comment]
I'm just stuck on the other components in v3 being built out and integrated first.

**Comment by jamesrashford** [comment]
I absolutely love the look and design of this. Is there anywhere I can integrate into my current set up?

**Comment by merwok** [comment]
This is an accepted request
